### PR TITLE
[cppcheck] Remove some redundant assignments.

### PR DESCRIPTION
**Pull request by AnilBK**
Remove **some** redundant Assignments from #51151.

https://godotcppcheck.000webhostapp.com/ and just check `redundantAssignment`

### CODE REVIEW

File: drivers/vulkan/rendering_device_vulkan.cpp [hunk 1]
```diff
@@ -7957,13 +7957,11 @@ void RenderingDeviceVulkan::compute_list_bind_uniform_set(ComputeListID p_list,
 
 				textures_to_storage[i]->used_in_compute = false;
 				textures_to_storage[i]->used_in_raster = false;
-				textures_to_storage[i]->used_in_compute = false;
```
Comment (pycbouh) [inline drivers/vulkan/rendering_device_vulkan.cpp:7960]: Maybe it's supposed to be `used_in_transfer`?

Comment (AnilBK) [reply drivers/vulkan/rendering_device_vulkan.cpp:7960]: Seems like. Asked in /rendering.

Comment (AnilBK) [reply drivers/vulkan/rendering_device_vulkan.cpp:7960]: Done.

File: modules/gdscript/gdscript_parser.cpp [hunk 2]
```diff
@@ -1117,7 +1117,6 @@ GDScriptParser::EnumNode *GDScriptParser::parse_enum() {
 				}
 				item.custom_value = value;
 			}
-			item.rightmost_column = previous.rightmost_column;
```
Comment (groud) [inline modules/gdscript/gdscript_parser.cpp:1120]: This looks like a false positive no?

Comment (AnilBK) [reply modules/gdscript/gdscript_parser.cpp:1120]: https://github.com/godotengine/godot/blob/2d446771d6701bb77432b6df5a768e53c9a6c95d/modules/gdscript/gdscript_parser.cpp#L1094-L1121

The line above(line 1120) is same as line 1095.
And,it in between them, item.rightmost_column isn't changed.previous.rightmost_column isn't modified either.

Comment (groud) [reply modules/gdscript/gdscript_parser.cpp:1120]: Ah my bad, it looks like github's formatting tricked me.

**Review by groud — APPROVED** [review]

**Pull request merged**